### DISCUSSION TITLE
Custom User-Agent when checking sites

### DISCUSSION THREAD

**Discussion by mochaaP** [upptime/upptime#331] [Ideas] [answered]
<https://github.com/upptime/uptime-monitor/blob/master/src/helpers/request.ts#L23>

When a `User-Agent` blocking strategy is configured on the site (e.g. scrawlers blocking), upptime will report a false downtime.
Setting headers manually in config won't resolve this problem, as the `User-Agent` string was set through `CURL_SETOPT`, which have a higher priority than headers.

Example regex that will filter `Koj Bot` out:
```regex
/Bot|Spider|BingPreview|Feedfetcher-Google|ia_archiver|MBCrawler|Mediapartners-Google|MSNot-media|Teoma|Yahoo! Slurp|^$/i
```

**Comment by AnandChowdhary** [comment, marked as answer]
Can you open an issue for this instead of an idea discussion? I think we should tread (and fix!) this as a bug. 😄

**Reply by ririko5834** [reply]
You should add config option to set custom user-agent
